src/gp_instance.cpp

```diff
@@ -37,11 +58,34 @@ void GPInstance::MakeEngine() {
   CheckSequencesAndTreesLoaded();
   ProcessLoadedTrees();
   SitePattern site_pattern(alignment_, tree_collection_.TagTaxonMap());
-  engine_ = std::make_unique<GPEngine>(
-      // To count GPCSSs, we add the usual suspects to the number of leaves (which are
-      // the fake PCSS).
-      site_pattern, sbn_parameters_.size() + tree_collection_.TaxonCount(),
-      mmap_file_path_);
+
+  ConstructDAG();
+
+  MakeGPEngine();
+  BuildPCSPIndexer();
+
+  rootward_order_ = RootwardPassTraversal();
+  leafward_order_ = LeafwardPassTraversal();
+
+  InitializeGPEngine();
+}
+
+void GPInstance::MakeGPEngine() {
+  SitePattern site_pattern(alignment_, tree_collection_.TagTaxonMap());
+
+  // Get number of parameters involving fake subsplits.
```

Comment:
This seems like a reasonable thing to factor out into a method.

Reply:
Factor out `SitePattern site_pattern(alignment_, tree_collection_.TagTaxonMap());`??